### PR TITLE
Mentee page!!

### PR DESCRIPTION
**Pull request by winterberry123**
i hope this works lmao :pp
can add a child associated with a mentee
Screen Shot 2023-04-24 at 8 29 20 PM

each child has 4 default folders - images, videos, links, flyers (pdf's)
Screen Shot 2023-04-24 at 8 29 54 PM

upload button on each mentee page allows you to upload files/links to selected folders (if none selected, it only displays on the "recent uploads")
Screen Shot 2023-04-24 at 8 31 21 PM

everytime you upload a file/link, it automatically sorts it into their respective default folders (.jpg -> images, .mov -> videos, etc).

database schema:
each profile (the user) has a "mentees" array containing all of their mentees. each mentee is their own document in the "mentees" collection, and each mentee document has a folders collection (each folder is a document with a files array field that contains links of all the media in that folder). so when you add a child on the mentees page, that creates a new document in the "mentees" collection & adds that doc's ID to the profile's "mentees" array. CHECK FOTC 4 FIREBASE FOR THIS :D:D (idk if i should screenshot for privacy issues rah)

BUG: DOES NOT WORK IF YOU SIGN UP WITH A NEW PROFILE AND DON'T LOG OUT AND LOG BACK IN. I THINK IT'S A SIGNUP ISSUE -- LIKE WHEN YOU SIGNUP YOU SHOULD BE LOGGED OUT, NOT IMMEDIATELY LOGGED IN??

### CODE REVIEW

File: client/src/pages/Mentees.jsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
+import React, { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import styles from '../styles/Mentees.module.css';
+import * as api from '../api';
+// import MenteeImage from '../assets/images/empty_mentees.svg';
```
Comment (xu-jerry) [inline client/src/pages/Mentees.jsx:10]: Were you going to use this for when there are no mentees? I can also push this to a later sprint if u didn't intend to implement it!

File: client/src/pages/ExpandedMentee.jsx [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+import PropTypes from 'prop-types';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import styles from '../styles/Mentees.module.css';
+import VideoIcon from '../assets/icons/videos_icon.svg';
+import ImageIcon from '../assets/icons/images_icon.svg';
+import FlyerIcon from '../assets/icons/flyers_icon.svg';
+import { storage } from './firebase';
```
Comment (xu-jerry) [inline client/src/pages/ExpandedMentee.jsx:13]: Can you change this to an api call just so we have less to do when we clean up?

**Review by xu-jerry — CHANGES_REQUESTED** [review]
I think this is the largest sprint and PR i have ever reviewed LMAO Great job!!! Only some small changes, and we should be ready to merge!

File: client/src/pages/Mentees.jsx (hunk 1 above)
Comment (winterberry123) [reply client/src/pages/Mentees.jsx:10]: ya i had it show up when there's no mentees with this line:
`{(mentees.length === 0) && (mentees)}`

but it was kinda glitchy; so whenever you refresh the mentees page the image would appear for a split second before the mentee cards are displayed. wasn't sure if that was an issue or not, so i just ultimately left it out

File: client/src/pages/ExpandedMentee.jsx (hunk 2 above)
Comment (winterberry123) [reply client/src/pages/ExpandedMentee.jsx:13]: should i move the entire "addMedia" function to the server, or just this part:
`const storageRef = ref(storage, `/images/${fileName}`);
      uploadBytes(storageRef, files).then((snapshot) => {
        getDownloadURL(snapshot.ref).then((url)` ?

cause the addMedia function sets a lotta states on the expanded mentee & media pages.

**Review by xu-jerry — APPROVED** [review]
SLAY thank you Joyce

**Pull request merged**